ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -38,28 +41,28 @@ import           Ouroboros.Network.Protocol.Handshake.Version
 import           Ouroboros.Network.ErrorPolicy
 import           Ouroboros.Network.IOManager
 import           Ouroboros.Network.Mux
-import           Ouroboros.Network.NodeToClient ( NodeToClientVersion (..)
-                                                , NodeToClientVersionData )
+import           Ouroboros.Network.NodeToClient (NodeToClientVersion (..),
+                     NodeToClientVersionData)
 import qualified Ouroboros.Network.NodeToClient as NodeToClient
-import           Ouroboros.Network.NodeToNode ( NodeToNodeVersion (..)
-                                              , NodeToNodeVersionData
-                                              , AcceptedConnectionsLimit (..)
-                                              , AcceptConnectionsPolicyTrace (..)
-                                              , DiffusionMode (..)
-                                              , RemoteAddress
-                                              )
-import qualified Ouroboros.Network.NodeToNode   as NodeToNode
-import           Ouroboros.Network.Socket ( ConnectionId (..)
-                                          , NetworkMutableState
-                                          , newNetworkMutableState
-                                          , cleanNetworkMutableState
-                                          , NetworkServerTracers (..)
-                                          )
-import           Ouroboros.Network.Subscription.Ip
+import           Ouroboros.Network.NodeToNode
+                     (AcceptConnectionsPolicyTrace (..),
+                     AcceptedConnectionsLimit (..), DiffusionMode (..),
+                     NodeToNodeVersion (..), NodeToNodeVersionData,
+                     RemoteAddress)
+import qualified Ouroboros.Network.NodeToNode as NodeToNode
+import           Ouroboros.Network.Socket (ConnectionId (..),
+                     NetworkMutableState, NetworkServerTracers (..),
+                     cleanNetworkMutableState, newNetworkMutableState)
 import           Ouroboros.Network.Subscription.Dns
+import           Ouroboros.Network.Subscription.Ip
 import           Ouroboros.Network.Subscription.Worker (LocalAddresses (..))
 import           Ouroboros.Network.Tracers
 
+data SocketTrace = SocketTrace
+  { socketTraceMessage :: Text
+  , socketTracePath    :: Maybe FilePath
+  } deriving (Eq, Show)
```

Comment:
The name is misleading, what about `DiffusionInitiatilizationTracer`, a bit long but self explaining (which will be important in a config file).

I think it's better to use a sum type instead of bare `Text`.  Then just looking at the constructors one can see what output it can generate.  A more detailed textuual representation should be defined in `cardano-node` (in the `Cardano.Tracing.OrphanedInstances.Network`).